flytekit/models/literals.py

```diff
@@ -912,6 +918,14 @@ def hash(self):
     def hash(self, value):
         self._hash = value
 
+    @property
+    def metadata(self):
+        """
+        This value holds metadata about the literal.
+        :rtype: typing.Dict[str, str]
+        """
+        return self._metadata
```

Comment:
```suggestion
return getattr(self, '_metadata', None)
```

fixes https://github.com/flyteorg/flytekit/pull/2147#issuecomment-1985257645

```pytb
AttributeError: 'Literal' object has no attribute '_metadata'
```

but I would imagine there is a better place to enforce initialization